src/api/mod.rs

```diff
@@ -186,7 +186,7 @@ pub trait Peripheral: Send + Sync + Clone + Debug {
 
     /// Returns the set of properties associated with the peripheral. These may be updated over time
     /// as additional advertising reports are received.
-    async fn properties(&self) -> Result<PeripheralProperties>;
+    async fn properties(&self) -> Result<Option<PeripheralProperties>>;
```

Comment:
If you don't know the properties yet, can't you just use `PeripheralProperties::default()`? The fields (other than the address) are all either optional or can be empty or 0 anyway.

Reply:
My rationale behind this was to honestly report "we don't know yet" instead of "it's empty". Though, now that I look more closely at `PeripheralProperties`, I do see the `has_scan_response` field, which I guess reports this already. Still, in order to conform to Rust conventions, I'd argue in favor of making `PeripheralProperties` an `Option` and removing the `has_scan_response` field.

Reply:
I'm in agreement on Option versus default. I'd rather know the surrounding state than just getting back something with nothing in it and not knowing if that's what the device reported.